### ISSUE TITLE
Issue with addToDate function

### ISSUE THREAD

**Issue by pdrvsky** [rehookify/datepicker#10]
Hi, at first I would like to thank you very much for your work. I love the datepicker lib! But unfortunately I've found a bug when it comes to navigation between months when the context date has the day number bigger then month before. To clarify, we have a datepicker with months [-1, 0] and let's say it's 31/01. Then we obtain correctly 2 calendars - December and January. But if we navigate back one month there will be a problem - we get December and December. It's happening because of faulty addToDate implementation - the day number is passed to the "new" month and if it's "overflowing" (i.e. there's no 31st of November) than the JS Date will increment the month, thus giving us 1st of December. The correct implementation can be seen in date-fns library (addMonths function): https://github.com/date-fns/date-fns/blob/main/src/addMonths/index.ts

**Comment by Feshchenko** [comment]
@pdrvsky, hi. 
By design, calendars are not sorted -> https://github.com/rehookify/datepicker/blob/main/src/utils/config.ts#L51
The date picker have DEFAULT_CALENDARS_CONFIG where offset is set to [`[0]`](https://github.com/rehookify/datepicker/blob/main/src/constants.ts#L33)
**calendars[0]** - will always point to the current month and after it you will get other offsets in order that you will pass to the config
The thing you need to change is to swap calendars in the UI.

**Comment by Feshchenko** [comment]
I read through your text one more time and it could happen because of poor `key` on the days.
Try to change key to something `key={d.$date.toString()}` [L36](https://codesandbox.io/p/sandbox/range-with-context-hooks-fyv72j?file=%2Fsrc%2Fcomponents%2Fcalendar.tsx)

**Comment by Feshchenko** [comment]
@pdrvsky, LMK if 👆helps you.

**Comment by pdrvsky** [comment]
Hi @Feshchenko, I've repeated all the steps you mentioned when I did the investigation by myself. What I really meant is the issue with the method `addToDate` itself. Let me explain this with an example.

1. You pass the date of 31st December to the mentioned method - the args will be obviously the date class instance (31.12.22), value: -1, part: 'month';
2. Then according to implementation the method creates another Date instance but changes only the 'month' part, so the Date constructor will be literally like this new Date(2022,10,31)
3. The date will be `Thu Dec 01 2022 00:00:00 GMT+0100 (Central European Standard Time)` as the JS assumed the 31st of November is actually a December 1st 🙃
